src/pybind/mgr/dashboard/frontend/src/app/shared/api/dashboard.service.spec.ts

```diff
@@ -1,23 +1,43 @@
-import { HttpClientModule } from '@angular/common/http';
-import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
-import { inject, TestBed } from '@angular/core/testing';
+import { HttpClient } from '@angular/common/http';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { fakeAsync, TestBed, tick } from '@angular/core/testing';
 
-import { appendFile } from 'fs';
+import 'rxjs/add/observable/of';
+import { Observable } from 'rxjs/Observable';
 
 import { DashboardService } from './dashboard.service';
 
 describe('DashboardService', () => {
+  let service: DashboardService;
+  let httpClient: HttpClient;
+  let count = 0;
+
   beforeEach(() => {
     TestBed.configureTestingModule({
       providers: [DashboardService],
-      imports: [HttpClientTestingModule, HttpClientModule]
+      imports: [HttpClientTestingModule]
     });
+
+    service = TestBed.get(DashboardService);
+    httpClient = TestBed.get(HttpClient);
+    count = 0;
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
   });
 
   it(
-    'should be created',
-    inject([DashboardService], (service: DashboardService) => {
-      expect(service).toBeTruthy();
+    'should call getHealth',
```

Comment:
This test as many others in other files can be shorten to something like the following:
```
  it('should call getHealth', () => {
    let path; // Outside to make sure the inside is called
    spyOn(httpClient, 'get').and.callFake((path2) => {
      path = path2;
    });
    service.getHealth(); // Call it - no need to test that 'http' gives back an observable
    expect(path).toBe('api/dashboard/health');
  });
```

Reply:
I agree with removing the count, since testing the path has the same result.

But I'm not sure about removing the return of Observable, since we are not testing what http returns, but what the method returns, and in some situations they differ.

And to keep the api tests more similar, we should enforce always testings the path of the http requests and the outcome of the service method.